src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { Element } from './types/Element';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleChange = (event: React.ChangeEvent<Element>) => {
+    const { id, value } = event.target;
+
+    switch (id) {
+      case 'titleInputForm':
+        setTitleError(false);
+        setTodoTitle(value);
+        break;
+
+      case 'userSelectForm':
+        setUserError(false);
+        setSelectedUser(getUserById(+(value)));
+        break;
+
+      default:
+        throw new Error('Unknown form type');
+    }
+  };
+
+  const handleAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setTitleError(true);
+    }
+
+    if (!selectedUser) {
+      setUserError(true);
+    }
+
+    if (!todoTitle || !selectedUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...visibleTodos.map(todo => todo.id)) + 1,
```

Comment:
move this to the variable
Also, you can create a separate function to get max id
```suggestion
      id: ...,
```